nemo_skills/pipeline/utils/mounts.py

```diff
@@ -88,9 +88,28 @@ def check_mounts(
     # Compute directory of all files
     remote_dir_list = []
 
+    # Get environment variables from cluster config to resolve any env var mounts
+    # Maybe can be optimized in the future to not do this every time
+    env_vars = None
+
     # Check paths and add to mount list if not mounted
     if check_mounted_paths:
         for path, default_mount in mount_map.items():
+            # Check if path contains ${} env variables placeholders
+            if path is not None and "${" in path and "}" in path:
+                if env_vars is None:
+                    env_vars = get_env_variables(cluster_config)
+
+                # Replace ${VAR} with the value from env_vars
+                for var in env_vars:
+                    if var not in env_vars:
+                        raise ValueError(
+                            f"Cluster config item `{path}` contains env variable placeholders, but the required env var is "
+                            f"not provided in the cluster config or environment to resolve."
+                        )
+
+                    path = path.replace(f"${{{var}}}", env_vars[var])
```

Comment:
Doesn't `os.path.expandvars` already handle both `$VAR` and `${VAR}`? https://docs.python.org/3/library/os.path.html#os.path.expandvars

Reply:
`env_vars` can be a different from os.environ. These are only those variables that are explicitly listed in cluster config, not all variables in the environment.

Reply:
I've updated the logic to use os.path.expandvars() and injected the value of `env_vars` inside `os.environ` in the `get_env_variables()` function